src/ui/wxWidgets/addeditpropsheet.cpp

```diff
@@ -1229,276 +1229,345 @@ static short GetSelectedDCA(const wxComboBox *pcbox,
+                      m_symbols    != m_item.GetSymbols().c_str()     ||
+                      oldPWP       != pwp);
+
+
+          if (!m_item.IsAlias()) {
+            bIsPSWDModified = (password != m_item.GetPassword());
           }
           else {
-            bIsPSWDModified = true;
+            // Update password to alias form
+            // Show text stating that it is an alias
+            const CItemData *pbci = m_core.GetBaseEntry(&m_item);
+            ASSERT(pbci);
+            if (pbci) {
+              StringX alias = L"[" +
+                  pbci->GetGroup() + L":" +
+                  pbci->GetTitle() + L":" +
+                  pbci->GetUser()  + L"]";
+              bIsPSWDModified = (password != alias);
+            }
+            else {
+              bIsPSWDModified = true;
+            }
           }
-        }
 
-      if (bIsModified) {
-        // Just modify all - even though only 1 may have actually been modified
+        if (bIsModified) {
+          // Just modify all - even though only 1 may have actually been modified
+          m_item.SetGroup(tostringx(group));
+          m_item.SetTitle(tostringx(m_title));
+          m_item.SetUser(m_user.empty() ?
+                        PWSprefs::GetInstance()->
+                        GetPref(PWSprefs::DefaultUsername).c_str() : m_user.c_str());
+          m_item.SetNotes(tostringx(m_notes));
+          m_item.SetURL(tostringx(m_url));
+          m_item.SetEmail(tostringx(m_email));
+          m_item.SetAutoType(tostringx(m_autotype));
+          m_item.SetRunCommand(tostringx(m_runcmd));
+          m_item.SetPWHistory(tostringx(m_PWHistory));
+
+          if (m_UseDatabasePolicyCtrl->GetValue()) {
+            // User has selected to use a named policy
+            wxString polName = m_cbxPolicyNames->GetValue();
+
+            // The default policy is neither item specific nor stored in the database
+            if (polName == _("Default Policy")) {
+              // Remove database policy information from item
+              m_item.SetPolicyName(tostringx(wxEmptyString));
+
+              // Remove item specific policy information from item
+              m_item.SetPWPolicy(tostringx(wxEmptyString));
+            }
+            // If it is not the default policy than it's a named policy from the database
+            else {
+              // Use policy that is stored in the database
+              m_item.SetPolicyName(tostringx(polName));
+
+              // Remove item specific policy information from item
+              m_item.SetPWPolicy(tostringx(wxEmptyString));
+            }
+          }
+          // User has selected to use an item specific policy
+          else {
+            // Use the data of the item specific policy collected from the UI
+            m_item.SetPWPolicy(pwp);
+
+            // Remove database policy information from item
+            m_item.SetPolicyName(tostringx(wxEmptyString));
+          }
```

Comment:
It's my own style opinion but this version is much more verbose than the previous.

Reply:
However, due to the additional new code, the resulting verbosity offers the advantage of more stability.

Reply:
I don't feel strongly enough about this and am dropping this comment.